File: b11047.cpp
#include <iostream>

int main()
{
	int coins_value[10];
	unsigned int num_coins, values;
	std::cin >> num_coins >> values;
	
	for (int i = 0; i < num_coins; ++i) {
		std::cin >> coins_value[i];
	}

	int result = 0;
	for (int i = num_coins - 1; i > -1; --i) {
		result += values / coins_value[i];
		values = values % coins_value[i];
		if (!values) {
			break;
		}
	}

	std::cout << result << std::endl;
}

File: b11047.py
coins_value = [0] * 10
num_coins, values = map(int, input().split())

for i in range(num_coins):
    coins_value[i] = int(input())

result = 0
for i in range(num_coins - 1, -1, -1):
    result += values // coins_value[i]
    values = values % coins_value[i]
    if not values:
        break
    
print(result)